src/test/java/teammates/sqlui/webapi/GetNotificationActionTest.java

```diff
@@ -0,0 +1,90 @@
+package teammates.sqlui.webapi;
+
+import static org.mockito.Mockito.when;
+
+import java.util.UUID;
+
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.Const;
+import teammates.storage.sqlentity.Notification;
+import teammates.ui.output.NotificationData;
+import teammates.ui.webapi.EntityNotFoundException;
+import teammates.ui.webapi.GetNotificationAction;
+import teammates.ui.webapi.InvalidHttpParameterException;
+import teammates.ui.webapi.JsonResult;
+
+/**
+ * SUT: {@link GetNotificationAction}.
+ */
+public class GetNotificationActionTest extends BaseActionTest<GetNotificationAction> {
+
+    @Override
+    String getActionUri() {
+        return Const.ResourceURIs.NOTIFICATION;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return GET;
+    }
+
+    @BeforeMethod
+    public void baseClassSetup() {
+        loginAsAdmin();
+    }
+
+    @Test
+    protected void testExecute_withValidNotificationId_shouldReturnData() {
+        NotificationAttributes notification = typicalBundle.notifications.get("notification1");
+        Notification testNotification = new Notification(
+                notification.getStartTime(),
+                notification.getEndTime(),
+                notification.getStyle(),
+                notification.getTargetUser(),
+                notification.getTitle(),
+                notification.getMessage());
+        notification.setNotificationId(testNotification.getId().toString());
+        NotificationData expected = new NotificationData(notification);
+
+        String[] requestParams = new String[] {
+                Const.ParamsNames.NOTIFICATION_ID, String.valueOf(testNotification.getId()),
+        };
+
+        when(mockLogic.getNotification(testNotification.getId())).thenReturn(testNotification);
+
+        GetNotificationAction action = getAction(requestParams);
+        JsonResult jsonResult = getJsonResult(action);
+
+        NotificationData output = (NotificationData) jsonResult.getOutput();
+        verifyNotificationEquals(expected, output);
+
+        when(mockLogic.getNotification(testNotification.getId())).thenCallRealMethod();
```

Comment:
Is the intended purpose of `thenCallRealMethod()` to reset the mock? Maybe can consider `reset(mockLogic)` because `thenCallRealMethod()` sets the real implementation to be called. But since `Logic.getNotification` should always be mocked, resetting is not strictly necessary.